src/robusta/core/sinks/robusta/robusta_sink.py

```diff
@@ -44,23 +48,29 @@ def __init__(
             self.cluster_name,
             signing_key,
         )
-        # start service discovery
+        # start cluster discovery
         self.__active = True
         self.__discovery_period_sec = DISCOVERY_PERIOD_SEC
         self.__services_cache: Dict[str, ServiceInfo] = {}
-        self.__thread = threading.Thread(target=self.__discover_services)
+        self.__nodes_cache: Dict[str, NodeInfo] = {}
+        self.__init_node_cache()
```

Comment:
The node cache seems to work differently than the services cache, in that the services cache isn't initialized here in the constructor. Can we keep the code consistent and have them work similarly?

Furthermore, if this throws (e.g. due to the supabase refresh bug) then service discovery will be entirely down until the runner starts.

Reply:
I opened another task to fix the service cache, so it will behave similar to the node cache.
The service cache calls read from the remote DB on every cycle which is redundant.

Re an error on the init, the refresh bug isn't an option here, but there might be another issue.
The discovery should be able to recover from that.
I'll fix it

Reply:
Sounds good.

Reply:
fixed